be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/SocialMediaHandler.scala

```diff
@@ -33,6 +36,27 @@ case object SocialMediaHandler extends MessageHandler {
     }
     case graphMessage@_ => graphMessage
   }
+
+  private final val unknownAnswerDatabase: String = "Database actor returned an unknown answer"
+
+  private def broadcastHelper(rpcMessage: JsonRpcRequest, broadcastData: Base64Data, broadcastChannel: Channel): GraphMessage = {
+    val askLaoData = (dbActor ? DbActor.ReadLaoData(rpcMessage.getParamsChannel))
+    Await.result(askLaoData, duration) match {
+      case DbActor.DbActorReadLaoDataAck(Some(laoData)) => {
+        val broadcastSignature: Signature = laoData.privateKey.signData(broadcastData)
+        val broadcastId: Hash = Hash.fromStrings(broadcastData.toString, broadcastSignature.toString)
+        val broadcastMessage: Message = Message(broadcastData, laoData.publicKey, broadcastSignature, broadcastId, List.empty)
```

Comment:
Why List.empty? Should it be laoData.witnesses instead?

Reply:
yes indeed!

Reply:
I'm not completely sure about how the witnesses work, as in CreateLao we get a list of witnesses' PublicKeys (which is stored inside LaoData) but the Message requires a list of WitnessSignaturePairs, I therefore left this as empty as the signing itself is not implemented. I should put a FIXME though